src/Features/CSharp/Portable/CodeRefactorings/InlineMethod/CSharpInlineMethodRefactoringProvider.cs

```diff
@@ -0,0 +1,157 @@
+                    return arrowExpressionNodes.Length == 1;
+                }
+            }
+
+            return false;
+        }
+
+        protected override IParameterSymbol? GetParameterSymbol(SemanticModel semanticModel, SyntaxNode argumentSyntaxNode, CancellationToken cancellationToken)
+            => argumentSyntaxNode is ArgumentSyntax argumentSyntax
+                ? argumentSyntax.DetermineParameter(semanticModel, allowParams: true, cancellationToken)
+                : null;
+
+        protected override SyntaxNode GenerateLiteralExpression(ITypeSymbol typeSymbol, object? value)
+            => ExpressionGenerator.GenerateExpression(typeSymbol, value, canUseFieldReference: false);
+
+        protected override bool IsExpressionSyntax(SyntaxNode syntaxNode)
+            => syntaxNode is ExpressionSyntax;
+
+        protected override string GetIdentifierTokenTextFromIdentifierNameSyntax(SyntaxNode syntaxNode)
+        {
+            if (syntaxNode is IdentifierNameSyntax identifierNameSyntax)
+            {
+                return identifierNameSyntax.Identifier.ValueText;
+            }
+
+            return string.Empty;
+        }
+
+        protected override SyntaxNode GenerateArrayInitializerExpression(ImmutableArray<SyntaxNode> arguments)
+            => SyntaxFactory.InitializerExpression(SyntaxKind.ArrayInitializerExpression, SyntaxFactory.SeparatedList(arguments));
+
+        protected override SyntaxNode GenerateLocalDeclarationStatementWithRightHandExpression(
+            string identifierTokenName,
+            ITypeSymbol type,
+            SyntaxNode expression)
+            => SyntaxFactory.LocalDeclarationStatement(
+                SyntaxFactory.VariableDeclaration(
+                    type.GenerateTypeSyntax(),
+                    SyntaxFactory.SingletonSeparatedList(
+                        SyntaxFactory.VariableDeclarator(
+                            SyntaxFactory.Identifier(identifierTokenName),
+                            argumentList: null,
+                            initializer: SyntaxFactory.EqualsValueClause((ExpressionSyntax)expression)))));
+
+        protected override SyntaxNode GenerateLocalDeclarationStatement(string identifierTokenName, ITypeSymbol type)
+            => SyntaxFactory.LocalDeclarationStatement(
+                SyntaxFactory.VariableDeclaration(
+                    type.GenerateTypeSyntax(),
+                    SyntaxFactory.SingletonSeparatedList(SyntaxFactory.VariableDeclarator(identifierTokenName))));
+
+        protected override SyntaxNode GenerateIdentifierNameSyntaxNode(string name)
+            => SyntaxFactory.IdentifierName(name);
+
+        protected override SyntaxNode? GetInlineStatement(SyntaxNode calleeMethodDeclarationSyntaxNode)
+        {
+            SyntaxNode? inlineSyntaxNode = null;
+            if (calleeMethodDeclarationSyntaxNode is MethodDeclarationSyntax declarationSyntax)
```

Comment:
shoulbe able to handle any member type.  my recommendation would be to take in teh block/arrow node instead.

Reply:
I though this is the same discussion we have above, so I will leave this resolved and we can have discussion in the other comment